packages/next/build/webpack-config.ts

```diff
@@ -60,11 +60,24 @@ function getOptimizedAliases(isServer: boolean): { [pkg: string]: string } {
   }
 
   const stubWindowFetch = path.join(__dirname, 'polyfills', 'fetch.js')
+  const stubWindowObjectAssign = path.join(
+    __dirname,
+    'polyfills',
+    'object-assign.js'
+  )
+
   return {
     __next_polyfill__fetch: require.resolve('whatwg-fetch'),
     unfetch$: stubWindowFetch,
     'isomorphic-unfetch$': stubWindowFetch,
     'whatwg-fetch$': stubWindowFetch,
+
+    // Alias Object.assign
+    '__next_polyfill_object-assign': require.resolve('object-assign'),
+    'object-assign$': stubWindowObjectAssign,
+    'object.assign$': stubWindowObjectAssign,
```

Comment:
Unfortunately, this polyfill is non-standard. We need to handle more cases.

```js
let a = require('object.assign');

// non-standard `getPolyfill` fn (returns `Object.assign`):
> a.getPolyfill() === a
true

// non-standard `implementation` prop (we must set to `Object.assign`)
> a.implementation
[Function: assign]

// non-standard `shim` fn (we can make it return `Object.assign`):
> a.shim()
[Function: assign]
```

Reply:
My suggestion would be to alias `object.assign` to a custom shim. Here's the implementation:

```js
// polyfills/variants/object.assign.js
function g() { return Object.assign; }
module.exports = {
  implementation: g(),
  shim: g,
  getPolyfill: g
};
```

Reply:
The `implementation` field above needs to be `Object.assign`, not `g`, correct @developit?

Reply:
Wouldn't this decorate the global instance with these fields too? 🤔

Reply:
`implementation` is a getter, so `module.exports.implementation` invokes `g()` to return `Object.assign`.

This does pollute the global instance, yes. The only way around this would be to return a proxy function:

```
// polyfills/variants/object.assign.js
var assign = Object.assign.bind(Object);
function g() { return assign; }
Object.defineProperties(g(), {
  implementation: { get: g },
  shim: { value: g },
  getPolyfill: { value: g }
});
module.exports = g();
```